src/main/java/hudson/plugins/sidebar_link/ProjectLinks.java

```diff
@@ -55,13 +55,17 @@ public ProjectLinks(List<LinkAction> links) {
 
     public List<LinkAction> getLinks() { return links; }
 
-    @Override
     public Collection<? extends Action> getJobActions(AbstractProject<?,?> job) {
         if(links == null)
             return new ArrayList<LinkAction>();
         return links;
     }
 
+    @Override
+    public Collection<? extends Action> getJobActions(Job<?,?> job) {
+        return Collections.EMPTY_SET;
```

Comment:
Adding `LinkAction`s through `getJobActions()` works for only `AbstractProject`s. And adding `LinkAction`s through the `TransientActionFactory` works for all `Job`s. 

So leaving this method as-is would render 2 links for a `FreeStyleProject` and 1 link for a `WorkflowJob`.

I think the correct solution here is have `getJobActions(Job<?,?> job)` delegate to `getJobActions(AbstractProject<?,?> job)` for `AbstractProject`s and be a noop in the `TransientActionFactory`